contrib/nydus-snapshotter/pkg/filesystem/nydus/fs.go

```diff
@@ -186,7 +190,19 @@ func (fs *filesystem) Umount(ctx context.Context, mountPoint string) error {
 	}
 
 	id := filepath.Base(mountPoint)
-	return fs.manager.DestroyBySnapshotID(id)
+	daemon, err := fs.manager.GetBySnapshotID(id)
+	if err != nil {
+		return err
+	}
+	if err := fs.manager.DestroyDaemon(daemon); err != nil {
+		return errors.Wrap(err, "destroy daemon err")
+	}
+	if err := fs.cacheMgr.DelSnapshot(daemon.ImageID); err != nil {
+		return errors.Wrap(err, "del snapshot err")
+	}
+	log.L.Debugf("remove snapshot %s\n", daemon.ImageID)
+	fs.cacheMgr.SchedGC()
```

Comment:
blob cache management is in snapshot Prepare() and Remove() interfaces, and call nydus/fs.go Mount()/Umount(), because blob cache management is happened when image.boot snapshot create and destroy.

Reply:
Ah, I see. I misunderstood it. Thanks.